cleanlab/regression/rank.py

```diff
@@ -0,0 +1,186 @@
+
+    method : {"residual", "outre"}, default="outre"
+        String specifying which method to use for scoring the quality of each label and identifying which labels appear most noisy.
+
+    Returns
+    -------
+    label_quality_scores:
+        Array of shape ``(N, )`` of scores between 0 and 1, one per example in the dataset.
+
+        Lower scores indicate examples more likely to contain a label issue.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from cleanlab.regression.rank import get_label_quality_scores
+    >>> labels = np.array([1,2,3,4])
+    >>> predictions = np.array([2,2,5,4.1])
+    >>> label_quality_scores = get_label_quality_scores(labels, predictions)
+    >>> label_quality_scores
+    array([0.00323821, 0.33692597, 0.00191686, 0.33692597])
+    """
+
+    # Check if inputs are valid
+    labels, predictions = assert_valid_prediction_inputs(
+        labels=labels, predictions=predictions, method=method
+    )
+
+    scoring_funcs: Dict[str, Callable[[np.ndarray, np.ndarray], np.ndarray]] = {
+        "residual": get_residual_score_for_each_label,
+        "outre": get_outre_score_for_each_label,
+    }
+
+    scoring_func = scoring_funcs.get(method, None)
+    if not scoring_func:
+        raise ValueError(
+            f"""
+            {method} is not a valid scoring method.
+            Please choose a valid scoring technique: {scoring_funcs.keys()}.
+            """
+        )
+
+    # Calculate scores
+    label_quality_scores = scoring_func(labels, predictions)
+    return label_quality_scores
+
+
+def get_residual_score_for_each_label(
+    labels: np.ndarray,
+    predictions: np.ndarray,
+) -> np.ndarray:
+    """Returns a residual label-quality score for each example.
+
+    This is function to compute label-quality scores for regression datasets,
+    where lower score indicate labels less likely to be correct.
+
+    Residual based scores can work better for datasets where independent variables
+    are based out of normal distribution.
+
+    Parameters
+    ----------
+    labels: np.ndarray
+        Labels in the same format expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    predictions: np.ndarray
+        Predicted labels in the same format expected by the :py:func:`get_label_quality_scores <cleanlab.regression.rank.get_label_quality_scores>` function.
+
+    Returns
+    -------
+    label_quality_scores: np.ndarray
+        Contains one score (between 0 and 1) per example.
+        Lower scores indicate more likely mislabled examples.
+
+    """
+    residual = predictions - labels
+    label_quality_scores = np.exp(-abs(residual))
+    return label_quality_scores
+
+
+def get_outre_score_for_each_label(
```

Comment:
can we convert this to private method? I don't see much advantage of offering this method in the public API